pkg/imagestore/ocilayout/ocilayout.go

```diff
@@ -0,0 +1,109 @@
+package ocilayout
+
+import (
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/pivotal/image-relocation/pkg/image"
+	"github.com/pivotal/image-relocation/pkg/registry"
+
+	"github.com/deislabs/duffle/pkg/imagestore"
+)
+
+type Builder struct {
+	registryClient registry.Client
+	archiveDir     string
+	logs           io.Writer
+}
+
+// ociLayout is an image store which stores images as an OCI image layout.
+type ociLayout struct {
+	layout registry.Layout
+	logs   io.Writer
+}
+
+func NewOciLayout() *Builder {
```

Comment:
Just curious why you chose not to return an `imagestore.Builder` here.

Reply:
That's mainly because it's idiomatic Go to return concrete types and leave any association with interfaces to the callers. Sometimes this approach runs out of steam as I think it did elsewhere in this PR and you _have_ to return an interface type in order for Go to match methods (typically in another interface), but generally returning concrete types is normal practice.

There are several articles on this topic, of which http://idiomaticgo.com/post/best-practice/accept-interfaces-return-structs/ isn't bad.

Reply:
BTW I came to Go from Java and I really didn't like this idiom when I first came across it, but I've since grown to accept it and even see the benefits. The only downside from my perspective is that it doesn't capture _intention_ quite so well in some situations.